src/drtsans/tof/eqsans/api.py

```diff
@@ -592,6 +607,8 @@ def pre_process_single_configuration(
         bkgd_ws_name = output_suffix + "_background"
         if not registered_workspace(bkgd_ws_name):
             bkgd_ws = prepare_data_workspaces(bkg_ws_raw, output_workspace=bkgd_ws_name, **prepare_data_conf)
+            if blocked_beam.data:
```

Comment:
subtracting the blocked beam from a run should be part of `prepare_data_workspaces`. In this function:
```
    if dark_current is not None and dark_current.data is not None:
        subtract_dark_current(output_workspace, dark_current.data)
    elif blocked_beam is not None and blocked_beam.data is not None:
        subtract_blocked_beam(output_workspace, dark_current.data)
```
and a dedicated `subtract_blocked_beam()` should be created.